wave_lang/kernel/wave/debug_log_hoist.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2025 The IREE Authors
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from .._support.tracing import CapturedTrace
+from ..lang.global_symbols import *
+from ..ops.wave_ops import (
+    get_custom,
+    Write,
+    Placeholder,
+)
+from .._support.dtype import DataType
+from .._support.indexing import IndexSymbol
+from typing import List, Tuple, TypedDict
+
+
+class DebugArgInfo(TypedDict):
+    symbol_name: str
+    debug_output_arg_id: int
+    dtype: DataType
+    symbolic_shape: Tuple[IndexSymbol, ...]
+
+
+def is_debug_log_transformer(node):
+    return get_custom(node).debug_log_transformer
```

Comment:
Can't we just do `isinstance`?

Reply:
Yes.  I was thinking I would be writing a family of these, but I can just switch to doing a series of `isinstance` calls.